primefaces/src/main/resources/META-INF/resources/primefaces/dialog/dialog.js

```diff
@@ -939,8 +939,18 @@ PrimeFaces.widget.ConfirmDialog = PrimeFaces.widget.Dialog.extend({
                 if(el.hasClass('ui-confirmdialog-yes') && PrimeFaces.confirmSource) {
                     var id = PrimeFaces.confirmSource.get(0);
                     var js = PrimeFaces.confirmSource.data('pfconfirmcommand');
-
-                    PrimeFaces.csp.executeEvent(id, js, e);
+                    
+                    // Define the pattern to match an empty function like: function(event){}
+                    const isEmptyFunctionPattern = /function\s*\(\s*event\s*\)\s*{\s*}/;
+                    // Test if the function matches the pattern
+                    if (isEmptyFunctionPattern.test(js)) {
```

Comment:
i wonder if it wouldnt be better to check js.contains(pf.ab || jsf.ajax)?

Reply:
see PF selenium:         return script.contains("PrimeFaces.ab(") || script.contains("pf.ab(")
                || script.contains("mojarra.ab(")
                || script.contains("myfaces.ab(")
                || script.contains("jsf.ajax.request") || script.contains("faces.ajax.request");

Reply:
Not a bad idea let me try that!